src/crab/ebpf_domain.cpp

```diff
@@ -1767,11 +1762,75 @@ void ebpf_domain_t::operator()(const Bin& bin) {
                         if (!bin.is64) {
                             input &= UINT32_MAX;
                         }
-                        uint64_t output = (int64_t)(input >> imm);
```

Comment:
I mean, regardless of any invariant in the code, it is by definition stored in a 64 bit register. The numerical domain, again by definition, tracks the signed interpretation of the underlying bits. If the interval happens to overflow, then either we take the modulo value (it is a singleton after all) or we assert it did not happen. I do not understand the condition.
(The same reasoning should apply to any interval range, but it might be trickier.)

It should not be an invariant of interval_t, since it is not designed for 64bit numbers exclusively, and handling offsets for example is different than numerical values.

I'm not going to press this matter in this PR, but it's important for me to understand and clarify what exactly we're doing here.

Reply:
My view: Allowing an interval like [X,Y] where (X > 0) and (INT64_MAX < Y < UINT64_MAX) can give greater precision than just using top(), since some operations (division, modulo, etc.) are defined to be unsigned.   Thus allowing interval_t to contain either a signed or an unsigned range of a 64-bit register can help give greater precision, where one just converts to top as needed for specific operations defined as signed or unsigned.

Reply:
Also lines 1759-1774 will go away if we accept PR #459.